### PR TITLE
Preserve data

### PR DESCRIPTION
**Pull request by ericmahare**
### In this section we did the following : -

- [ ] Preserved data to the json files after exiting the app

### CODE REVIEW

File: read_data.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      @my_app.people.push(teacher)
+    end
+  end || []
+end
+
+def read_rentals
+  file = File.read('rentals.json') if File.exist?('rentals.json')
+  rentals = JSON.parse(file) unless file.chomp.empty?
+  add_rental(rentals)
+end
```
Comment (medaminedev66) [inline read_data.rb]: - [ ] You should handle the case when one or more of the JSON files are missing. I tried to remove the `people..json, retals.json, books.json` files. When I run your app, it shows the following error:

image

Please make sure that if one of the files is missing, your app will create a new one with an empty array and save the data there when the app is exited.

**Review by medaminedev66 — CHANGES_REQUESTED** [review]
Hi @braincee and @ericmahare  😃, :wave:, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- The data is saved into JSON files after closing the app.
- all data is loaded from JSON files after starting the app.
Great job 👏

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by sja-thedude — APPROVED** [review]
## Project Approved 👍 

Hi @ericmahare & @braincee, This is @sja-thedude your code reviewer.

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

giphy

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by braincee** [comment]
Thanks very much, @sja-thedude, @medaminedev66 and @ericmahare 

thank-you-thankyou

**Pull request merged**